ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryBlobStorageService.java

```diff
@@ -798,9 +805,12 @@ public void onCompletion(final GetBlobResult routerResult, Exception routerExcep
                       BlobInfo blobInfo = routerResult.getBlobInfo();
                       Map<String, String> userMetadata = RestUtils.buildUserMetadata(blobInfo.getUserMetadata());
                       if (userMetadata == null) {
+                        // in which case we run here?????
```

Comment:
reminder to remove comment. This might be because we used to not have a set format for user metadata, so in those cases we will just return the bytes directly. @vgkholla can probably explain more.